lib/python/qmk/build_targets.py

```diff
@@ -118,7 +118,10 @@ def generate_compilation_database(self, build_target: str = None, skip_clean: bo
         self.prepare_build(build_target=build_target, **env_vars)
         command = self.compile_command(build_target=build_target, dry_run=True, **env_vars)
         from qmk.cli.generate.compilation_database import write_compilation_database  # Lazy load due to circular references
-        write_compilation_database(command=command, output_path=QMK_FIRMWARE / 'compile_commands.json', skip_clean=skip_clean, **env_vars)
+        output_path = QMK_FIRMWARE / 'compile_commands.json'
+        write_compilation_database(command=command, output_path=output_path, skip_clean=skip_clean, **env_vars)
+        if output_path.exists() and HAS_QMK_USERSPACE:
+            shutil.copy(str(output_path), str(QMK_USERSPACE / 'compile_commands.json'))
```

Comment:
`shutil.copy` should handle pathlib
```suggestion
            shutil.copy(output_path, QMK_USERSPACE / 'compile_commands.json')
```